test/RevokableTokenLock.spec.ts

```diff
@@ -0,0 +1,125 @@
+import {expect} from 'chai';
+import {BigNumber} from 'ethers';
+import {ethers, waffle} from 'hardhat';
+import {IERC20, RevokableTokenLock} from '../typechain';
+import {ONE, TWO, ZERO_ADDRESS, HOUR} from './shared/Constants';
+import {setNextBlockTimeStamp} from './shared/TimeManipulation';
+import {mineBlocks} from './shared/BlockManipulation';
+
+const {loadFixture} = waffle;
+
+let token: IERC20;
+let revokableTokenLock: RevokableTokenLock;
+
+describe('RevokableTokenLock', async () => {
+  const [owner, revoker, recipient, other] = waffle.provider.getWallets();
+  const amount = ethers.utils.parseEther('20');
+
+  // TODO: maybe create shared fixtures that can be imported by the test files
+  async function fixture() {
+    const TokenFactory = await ethers.getContractFactory('TestERC20');
+    const token = (await TokenFactory.deploy('TEST', 'TEST')) as IERC20;
+    const RevokableTokenLockFactory = await ethers.getContractFactory('RevokableTokenLock');
+    const revokableTokenLock = (await RevokableTokenLockFactory.deploy(
+      token.address,
+      revoker.address
+    )) as RevokableTokenLock;
+
+    const dt = Math.floor(new Date().getTime() / 1000);
+    const begin = ethers.BigNumber.from(dt).add(HOUR);
+    const cliff = begin.add(HOUR.mul(5));
+    const end = begin.add(HOUR.mul(20));
+    await revokableTokenLock.setupVesting(recipient.address, begin, cliff, end);
+    await token.approve(revokableTokenLock.address, amount);
+    await revokableTokenLock.lock(recipient.address, amount);
+
+    return {token, revokableTokenLock};
+  }
+  beforeEach('deploy fixture', async () => {
+    ({token, revokableTokenLock} = await loadFixture(fixture));
+  });
+  // TODO: maybe do getStorageAt() tests as well
```

Comment:
do we want to? what would these be used for? I prefer testing for functionality instead, not that important that a specific value at a specific storage slot is set

Reply:
getStorageAt() tests assumes that there's some error w/ how values are stored on the contract. This is definitely a nice to have but is non trivial to do and not as important now.